bmw_cd_exporter.py

```diff
@@ -113,25 +133,36 @@ def parse_args():
             exit(1)
         if args.interval < 1:
             args.interval = 1
-
-        # register & start prometheus exporter server
-        print('INFO: Registering Prometheus metrics...')
+        print(time.asctime(time.localtime(time.time())),":" 'INFO: Parse JSON File')
+           
         with open(args.attributes) as json_file:
             attrs = json.load(json_file)
+        
+        if (args.mqtt):
+            collectData(args.user, args.password, args.region, args.attributes, args.mqtt)
+            while True:
+                time.sleep(args.interval*60)
+                collectData(args.user, args.password, args.region, args.attributes, args.mqtt)
+        else:
+            # register & start prometheus exporter server
+        
+            print('INFO: Registering Prometheus metrics...')
             for attr in attrs:
                 gauges[attr['name']] = Gauge(app + '_' + attr['name'] + '_' + attr['unit'], attr['desc'], ['vehicle','vin'])
 
-        # Get initial metrics
-        print('INFO: Getting data from BMW Connected Drive...')
-        collectData(args.user, args.password, args.region, args.attributes)
-
-        # Start HTTP server
-        print('INFO: Starting HTTP server...')
-        start_http_server(port)
-        print("INFO: BMW Connected Drive exporter for Prometheus. Serving on port: {}".format(port))
-        while True: 
-            time.sleep(args.interval * 60)
-            collectData(args.user, args.password, args.region, args.attributes)
+            # Get initial metrics
+            print(time.asctime(time.localtime(time.time())), ":", 'INFO: Getting data from BMW Connected Drive...')
+            collectData(args.user, args.password, args.region, args.attributes, '')
+
+            # Start HTTP server
+            print('INFO: Starting HTTP server...')
+            start_http_server(port)
+            print("INFO: BMW Connected Drive exporter for Prometheus. Serving on port: {}".format(port))
+            while True: 
+                time.sleep(args.interval * 60)
+                collectData(args.user, args.password, args.region, args.attributes, '')
```

Comment:
This should be done in a way that even when an MQTT address is passed in, the HTTP server would still work.  It shouldn't be one or the other.

Reply:
I undertstand your position by why having an Http server up&running (with potential security risk) if we do not use the prometheus feature ?

Reply:
Good point!

In this case, it should enable MQTT when `--mqtt` is passed in, and it should enable HTTP server when `--port` is passed in. It should enable both, if both are passed in.